lib/extend/cask.rb

```diff
@@ -25,6 +25,7 @@ def self.installed_apps
           :auto_updates       => cask.auto_updates,
           :homepage           => cask.homepage,
           :installed_versions => versions,
+          :tap                => cask.tap.name,
```

Comment:
Thank you for your contribution! It seems that this not always work, I did a quick test and it failed

```
❯ brew cu --all                                                                                                                                                                     
==> Options
Include auto-update (-a): true
Include latest (-f): false
==> Updating Homebrew
Already up-to-date.
Error: undefined method `name' for nil:NilClass
/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/lib/extend/cask.rb:28:in `block in installed_apps'
/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/lib/extend/cask.rb:11:in `map'
/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/lib/extend/cask.rb:11:in `installed_apps'
/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/lib/bcu/command/upgrade.rb:17:in `process'
/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/lib/bcu.rb:18:in `process'
/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/cmd/brew-cu.rb:69:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/utils.rb:85:in `require?'
/usr/local/Homebrew/Library/Homebrew/brew.rb:121:in `<main>'
```

You probably need to add few checks?

Reply:
Out of curiosity, how did you get it to throw this error?

Reply:
Hey @404NetworkError, I haven't done anything special, just checkout out your branch and then run `brew cu --all` "as usual" and it failed with that error message.

Reply:
I can't test because I don't currently have any outdated apps, but I think this patch should fix the error:

```
From 8af112f951cd063b196357b6720502cb3bb05c2c Mon Sep 17 00:00:00 2001
From: Stephen Wade <stephen@stephenwade.me>
Date: Sat, 17 Oct 2020 16:15:54 -0400
Subject: [PATCH] fix undefined method 'name'

---
 lib/bcu/command/upgrade.rb | 3 ++-
 lib/extend/cask.rb         | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/bcu/command/upgrade.rb b/lib/bcu/command/upgrade.rb
index e65b589..d48bfc0 100644
--- a/lib/bcu/command/upgrade.rb
+++ b/lib/bcu/command/upgrade.rb
@@ -85,7 +85,8 @@ module Bcu

       begin
         # Force to install the latest version.
-        cmd = "brew reinstall #{options.install_options} #{app[:tap]}/#{app[:token]} --force " + verbose_flag
+        app_str = app[:tap].nil? ? app[:token] : "#{app[:tap]}/#{app[:token]}"
+        cmd = "brew reinstall #{options.install_options} #{app_str} --force " + verbose_flag
         success = system "#{cmd}"
       rescue
         success = false
diff --git a/lib/extend/cask.rb b/lib/extend/cask.rb
index d50f2a5..413e3ef 100644
--- a/lib/extend/cask.rb
+++ b/lib/extend/cask.rb
@@ -25,7 +25,7 @@ module Cask
           :auto_updates       => cask.auto_updates,
           :homepage           => cask.homepage,
           :installed_versions => versions,
-          :tap                => cask.tap.name,
+          :tap                => cask.tap&.name,
         }
       rescue CaskUnavailableError
         {
--
2.28.0
```

Reply:
Added your patch to the PR. I looks like it should work. I'm still unable to replicate the problem @ondrejfuhrer is having, even with no casks installed.

Reply:
All right, no more issues now after the patch, tested it with `dash` and worked as expected. Thank you both for your contribution! 🎖️